main.go

```diff
@@ -60,8 +63,15 @@ func main() {
 	flag.Parse()
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
+	cacheNamespaces := []string{namespace}
+	if namespace != argocdNamespace {
```

Comment:
This is to avoid creating two informers, two caches, etc. for the same namespace. Happy to change it with a more elegant solution.

Reply:
Could you give some context around the cache? Seems to be a new addition and not sure what it does.
Also, the argocdNamespace seems to only be used for this check?
Shouldn't we propagate this somewhere? or is it what the cache is doing?

Reply:
ok, I've read https://pkg.go.dev/sigs.k8s.io/controller-runtime#hdr-Clients_and_Caches and I understand a bit better now. Question: from my understanding we're moving from no cache to using cache. Is this going to cause issues in terms of cache invalidation?

Also, in the context where the controller runs in a different namespace than argocd, will it get notified by the resource changed on the appset or secrets? Sounds to me like the cache is only caching, not giving you the ability to watch another namespace?

Reply:
> Question: from my understanding we're moving from no cache to using cache. Is this going to cause issues in terms of cache invalidation?

My understanding from the link above is that when I read
```
The default client provided by the manager reads from a local shared cache (pkg/cache) and writes directly to the API server [...]
```
it means we had the cache even before.
Reading at https://book.kubebuilder.io/cronjob-tutorial/empty-main.html#every-journey-needs-a-start-every-program-a-main and https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/cache#MultiNamespacedCacheBuilder makes me think that instead of caching ProgressiveSync, Applications and Secrets across all the namespace we cache only the objects in those namespaces.

Reply:
>  Also, the argocdNamespace seems to only be used for this check?

Basically the idea is "if namespace and argocdNamespace are the same, watch only one namespace, otherwise the two of them"

Reply:
is adding a namespace as a cacheable space automatically makes it watch it? I would have thought it was under the namespace itself. You're the expert though :-)

Reply:
```
MultiNamespacedCacheBuilder - Builder function to create a new multi-namespaced cache. This will scope the cache to a list of namespaces. Listing for all namespaces will list for all the namespaces that this knows about. By default this will create a global cache for cluster scoped resource. Note that this is not intended to be used for excluding namespaces, this is better done via a Predicate. Also note that you may face performance issues when using this with a high number of namespaces. 
```
I would say yes.
I also added it into the tests and they're passing so it seems to me we can still watch the resources.

Reply:
Ah, I see your question now. Besides reading the docs, the only option is to add two namespaces to the tests: the random one + always "argocd". Then we add a test where we create a secret in argoCD and we make sure we see the event.

What do you think?

Reply:
yeah that works, or we could also decide/enforce the use of the same namespace as argocd and make it unique by design? or is this going to conflict with multiple argocd instances running in the same cluster?

Reply:
Unless we drop that requirement so when we update the controller is for everyone :)

Reply:
that's a strong change of approach but not necessarily a bad idea actually;
in practice the controller would be "bundled" with a running argocd controller so why not making this a strong 1:1 relationship.

How would you feel about that?
And everyone else?

Reply:
Happy with that at the moment. The AppSet controller has the same approach